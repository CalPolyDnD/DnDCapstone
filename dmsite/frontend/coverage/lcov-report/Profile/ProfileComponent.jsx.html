<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Profile/ProfileComponent.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">Profile</a> ProfileComponent.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.13% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">6.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">27.27% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/30</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { FormErrors } from './FormErrors';
import { withRouter } from 'react-router-dom';
import {
  Container, Card, Button, CardBody,
  Input, CardHeader,
} from 'reactstrap';
&nbsp;
class Profile extends React.Component {
&nbsp;
  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = {
      popoverOpen: false,
      username: '',
      email: '',
      password: '',
      formErrors: {email: '', password: '', username: ''},
      usernameValid: false,
      emailValid: false,
      passwordValid: false,
      formValid: false
    };
  }
&nbsp;
  handleUserInput = <span class="fstat-no" title="function not covered" >(e</span>) =&gt; {
    const name = <span class="cstat-no" title="statement not covered" >e.target.name;</span>
    const value = <span class="cstat-no" title="statement not covered" >e.target.value;</span>
<span class="cstat-no" title="statement not covered" >    this.setState({[name]: value},</span>
<span class="fstat-no" title="function not covered" >        ()</span> =&gt; { <span class="cstat-no" title="statement not covered" >this.validateField(name, value) </span>});
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  va</span>lidateField(fieldName, value) {
    let fieldValidationErrors = <span class="cstat-no" title="statement not covered" >this.state.formErrors;</span>
    let emailValid = <span class="cstat-no" title="statement not covered" >this.state.emailValid;</span>
    let passwordValid = <span class="cstat-no" title="statement not covered" >this.state.passwordValid;</span>
    let usernameValid = <span class="cstat-no" title="statement not covered" >this.state.usernameValid;</span>
&nbsp;
&nbsp;
<span class="cstat-no" title="statement not covered" >    switch(fieldName) {</span>
      case 'email':
<span class="cstat-no" title="statement not covered" >        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);</span>
<span class="cstat-no" title="statement not covered" >        fieldValidationErrors.email = emailValid ? '' : ' is invalid';</span>
<span class="cstat-no" title="statement not covered" >        break;</span>
      case 'password':
<span class="cstat-no" title="statement not covered" >        passwordValid = value.length &gt;= 6;</span>
<span class="cstat-no" title="statement not covered" >        fieldValidationErrors.password = passwordValid ? '': ' is too short';</span>
<span class="cstat-no" title="statement not covered" >        break;</span>
      case 'username':
<span class="cstat-no" title="statement not covered" >        usernameValid = value.length &gt;= 0;</span>
<span class="cstat-no" title="statement not covered" >        fieldValidationErrors.username = usernameValid ? '': ' Create a username';</span>
<span class="cstat-no" title="statement not covered" >        break;</span>
      default:
<span class="cstat-no" title="statement not covered" >        break;</span>
    }
<span class="cstat-no" title="statement not covered" >    this.setState({formErrors: fieldValidationErrors,</span>
                  emailValid: emailValid,
                  passwordValid: passwordValid,
                  usernameValid: usernameValid
                }, this.validateForm);
}
&nbsp;
<span class="fstat-no" title="function not covered" >  to</span>ggle() {
    const { popoverOpen } = <span class="cstat-no" title="statement not covered" >this.state;</span>
<span class="cstat-no" title="statement not covered" >    this.setState(<span class="fstat-no" title="function not covered" >pr</span>evState =&gt; (<span class="cstat-no" title="statement not covered" >{</span></span>
      ...prevState,
      popoverOpen: !popoverOpen,
    }));
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  va</span>lidateForm() {
<span class="cstat-no" title="statement not covered" >    this.setState({formValid: this.state.emailValid &amp;&amp; this.state.passwordValid &amp;&amp; this.state.usernameValid});</span>
  }
&nbsp;
  errorClass(error) {
    return(error.length === 0 ? '' : <span class="branch-1 cbranch-no" title="branch not covered" >'has-error')</span>;
  }
&nbsp;
  render() {
    return (
      &lt;Container fluid style={{ justifyContent: 'center' }}&gt;
        &lt;div style={{
          justifyContent: 'center', width: '100%', alignSelf: 'center', display: 'flex',
        }}
        &gt;
          &lt;Card style={{
            justifyContent: 'center', width: '40%', alignSelf: 'center', display: 'flex', borderWidth: 0
          }}
          &gt;
            &lt;CardHeader tag="h3" align="center" style={{ backgroundColor: '#303030', color: 'white' }}&gt;User Profile and Settings&lt;/CardHeader&gt;
            &lt;CardBody style={{ width: '100%', justifyContent: 'space-between', backgroundColor: '#3d3d3d', color: 'white' }} &gt;
              &lt;div className= "panel panel-default"&gt;
                &lt;FormErrors formErrors={this.state.formErrors} /&gt;
              &lt;/div&gt;
              &lt;div className={`form-group ${this.errorClass(this.state.formErrors.username)}`}&gt;
                &lt;label htmlFor="user"&gt;Username&lt;/label&gt;
                &lt;Input type="username" required className="form-control" name="username"
                       style={{ color: 'white', backgroundColor: '#303030', borderWidth: 0 }}
                       placeholder="User"
                       value={this.state.user}
                       onChange={this.handleUserInput}  /&gt;
              &lt;/div&gt;
              &lt;div className={`form-group ${this.errorClass(this.state.formErrors.email)}`}&gt;
                &lt;label htmlFor="email"&gt;Email&lt;/label&gt;
                &lt;Input type="email" className="form-control" name="email"
                       style={{ color: 'white', backgroundColor: '#303030', borderWidth: 0 }} placeholder="Example@ex.com"
                       value={this.state.email}
                       onChange={this.handleUserInput}  /&gt;
              &lt;/div&gt;
              &lt;div className={`form-group ${this.errorClass(this.state.formErrors.password)}`}&gt;
                &lt;label htmlFor="password"&gt;Password&lt;/label&gt;
                &lt;Input type="password" className="form-control" name="password"
                       style={{ color: 'white', backgroundColor: '#303030', borderWidth: 0 }} placeholder="Password123"
                       value={this.state.password}
                       onChange={this.handleUserInput}  /&gt;
              &lt;/div&gt;
              &lt;div&gt;
                &lt;label htmlFor="security"&gt;Security&lt;/label&gt;
                &lt;Input style={{ color: 'white', backgroundColor: '#303030', borderWidth: 0 }} placeholder="Basic User" /&gt;
              &lt;/div&gt;
              &lt;br /&gt;
            &lt;/CardBody&gt;
            &lt;div className="d-flex justify-content-center pt-2" style={{ backgroundColor: '#3d3d3d', color: 'white' }}&gt;
              &lt;Button color = "primary" size="md" className="btn-block mt-0" disabled={!this.state.formValid} onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; { <span class="cstat-no" title="statement not covered" >this.props.history.push('/home'); </span>}} &gt;Save&lt;/Button&gt;
              &lt;Button color="secondary" size="md" className="btn-block mt-0" onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; { <span class="cstat-no" title="statement not covered" >this.props.history.push('/home'); </span>}}&gt;Back&lt;/Button&gt;
            &lt;/div&gt;
          &lt;/Card&gt;
        &lt;/div&gt;
      &lt;/Container&gt;
    );
  }
}
&nbsp;
export default withRouter(Profile);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Feb 21 2019 10:47:18 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
