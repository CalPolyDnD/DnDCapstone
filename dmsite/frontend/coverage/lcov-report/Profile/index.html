<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Profile</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> Profile
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.84% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/38</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.84% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/37</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="FormErrors.js"><a href="FormErrors.js.html">FormErrors.js</a></td>
	<td data-value="80" class="pic high"><div class="chart"><div class="cover-fill" style="width: 80%;"></div><div class="cover-empty" style="width:20%;"></div></div></td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	</tr>

<tr>
	<td class="file low" data-value="ProfileComponent.jsx"><a href="ProfileComponent.jsx.html">ProfileComponent.jsx</a></td>
	<td data-value="30.3" class="pic low"><div class="chart"><div class="cover-fill" style="width: 30%;"></div><div class="cover-empty" style="width:70%;"></div></div></td>
	<td data-value="30.3" class="pct low">30.3%</td>
	<td data-value="33" class="abs low">10/33</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="31.25" class="pct low">31.25%</td>
	<td data-value="32" class="abs low">10/32</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Apr 30 2019 00:01:20 GMT-0700 (PDT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
