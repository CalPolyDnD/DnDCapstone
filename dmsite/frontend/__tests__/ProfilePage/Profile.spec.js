import 'jsdom-global/register';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { mount, configure } from 'enzyme';
import Profile from '../../src/Profile/ProfileComponent';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

describe('Layout-Footer', () => {
	it('should render the component without errors', () => {
		const wrapper = mount(
			<MemoryRouter>
				<Profile />
			</MemoryRouter>,
		);
		expect(toJson(wrapper.find(Profile))).toMatchSnapshot();
	});
})