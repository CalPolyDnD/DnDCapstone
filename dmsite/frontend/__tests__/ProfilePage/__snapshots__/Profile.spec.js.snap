// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout-Footer should render the component without errors 1`] = `
<withRouter(Profile)>
  <Route>
    <Profile
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "ztj4p6",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "ztj4p6",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "key": "ztj4p6",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
    >
      <Container
        fluid={true}
        style={
          Object {
            "justifyContent": "center",
          }
        }
        tag="div"
      >
        <div
          className="container-fluid"
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <div
            style={
              Object {
                "alignSelf": "center",
                "display": "flex",
                "justifyContent": "center",
                "paddingTop": "3%",
                "width": "100%",
              }
            }
          >
            <Card
              style={
                Object {
                  "alignSelf": "center",
                  "borderWidth": 0,
                  "display": "flex",
                  "justifyContent": "center",
                  "width": "40%",
                }
              }
              tag="div"
            >
              <div
                className="card"
                style={
                  Object {
                    "alignSelf": "center",
                    "borderWidth": 0,
                    "display": "flex",
                    "justifyContent": "center",
                    "width": "40%",
                  }
                }
              >
                <CardHeader
                  align="center"
                  style={
                    Object {
                      "backgroundColor": "#303030",
                      "color": "white",
                    }
                  }
                  tag="h3"
                >
                  <h3
                    align="center"
                    className="card-header"
                    style={
                      Object {
                        "backgroundColor": "#303030",
                        "color": "white",
                      }
                    }
                  >
                    Profile Settings
                  </h3>
                </CardHeader>
                <CardBody
                  style={
                    Object {
                      "backgroundColor": "#3d3d3d",
                      "color": "white",
                      "justifyContent": "space-between",
                      "width": "100%",
                    }
                  }
                  tag="div"
                >
                  <div
                    className="card-body"
                    style={
                      Object {
                        "backgroundColor": "#3d3d3d",
                        "color": "white",
                        "justifyContent": "space-between",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="panel panel-default"
                    >
                      <FormErrors
                        formErrors={
                          Object {
                            "password": "",
                          }
                        }
                      >
                        <div
                          className="formErrors"
                        />
                      </FormErrors>
                    </div>
                    <div
                      style={
                        Object {
                          "paddingBottom": "10%",
                        }
                      }
                    >
                      <label>
                        Username
                        <br />
                        <br />
                        ..................................
                      </label>
                    </div>
                    <div
                      style={
                        Object {
                          "paddingBottom": "10%",
                        }
                      }
                    >
                      <label>
                        Email
                        <br />
                        <br />
                        ..................................
                      </label>
                    </div>
                    <div
                      style={
                        Object {
                          "fontSize": "125%",
                          "paddingBottome": "5%",
                          "paddingLeft": "33%",
                        }
                      }
                    >
                      <label>
                        Change Password
                      </label>
                    </div>
                    <div
                      className="form-group "
                    >
                      <label
                        htmlFor="password1"
                      >
                        New Password
                      </label>
                      <Input
                        className="form-control"
                        name="password"
                        onChange={[Function]}
                        placeholder=""
                        style={
                          Object {
                            "backgroundColor": "#303030",
                            "borderWidth": 0,
                            "color": "white",
                          }
                        }
                        type="password"
                      >
                        <input
                          className="form-control form-control"
                          name="password"
                          onChange={[Function]}
                          placeholder=""
                          style={
                            Object {
                              "backgroundColor": "#303030",
                              "borderWidth": 0,
                              "color": "white",
                            }
                          }
                          type="password"
                        />
                      </Input>
                    </div>
                    <div
                      className="form-group "
                    >
                      <label
                        htmlFor="password2"
                      >
                        Confirm New Password
                      </label>
                      <Input
                        className="form-control"
                        name="password"
                        onChange={[Function]}
                        placeholder=""
                        style={
                          Object {
                            "backgroundColor": "#303030",
                            "borderWidth": 0,
                            "color": "white",
                          }
                        }
                        type="password"
                      >
                        <input
                          className="form-control form-control"
                          name="password"
                          onChange={[Function]}
                          placeholder=""
                          style={
                            Object {
                              "backgroundColor": "#303030",
                              "borderWidth": 0,
                              "color": "white",
                            }
                          }
                          type="password"
                        />
                      </Input>
                    </div>
                    <div
                      style={
                        Object {
                          "backgroundColor": "#3d3d3d",
                          "paddingLeft": "15%",
                          "paddingRight": "15%",
                        }
                      }
                    >
                      <Button
                        className="btn-block mt0 login-form-button"
                        color="primary"
                        onClick={[Function]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="btn-block mt0 login-form-button btn btn-primary"
                          onClick={[Function]}
                          type="button"
                        >
                          Save and Return
                        </button>
                      </Button>
                       
                    </div>
                  </div>
                </CardBody>
              </div>
            </Card>
          </div>
        </div>
      </Container>
    </Profile>
  </Route>
</withRouter(Profile)>
`;
